backend/src/utils/email_send.rs

```diff
@@ -0,0 +1,166 @@
+
+#[allow(non_snake_case)]
+#[derive(Serialize)]
+struct Template {
+    TemplateID: i32,
+    TemplateData: String,
+}
+#[allow(non_snake_case)]
+#[derive(Serialize)]
+struct Body {
+    FromEmailAddress: String,
+    ReplyToAddresses: String,
+    Destination: Vec<String>,
+    Template: Template,
+    Subject: String,
+}
+
+fn gen_auth(param: &Body, timestamp: String) -> String {
+    // 密钥参数
+    let secret_id = SECRET_ID;
+    let secret_key = SECRET_KEY;
+    let host = "ses.tencentcloudapi.com".to_string();
+    let date = Utc::now().format("%Y-%m-%d").to_string();
+    let service = "ses".to_string();
+    let algorithm = "TC3-HMAC-SHA256";
+    // step 1: 拼接规范请求串
+    let http_request_method = "POST".to_string();
+    let canonical_uri = "/".to_string();
+    let canonical_querystring = "".to_string();
+    let ct = "application/json; charset=utf-8".to_string();
+    let payload_str = serde_json::to_string(param).unwrap();
+    let payload_vec: Vec<&str> = payload_str.split(':').collect();
+    let payload = payload_vec.join(": ");
+    let payload_vec: Vec<&str> = payload.split(',').collect();
+    let payload = payload_vec.join(", ");
+    // println!("{}", payload);
+    let canonical_headers = format!("content-type:{}\nhost:{}\n", ct, host);
+    let signed_headers = "content-type;host".to_string();
+    let mut hasher = Sha256::new();
+    hasher.input_str(&payload);
+    let hashed_request_payload = hasher.result_str();
+    let canonical_request = format!(
+        "{}\n{}\n{}\n{}\n{}\n{}",
+        http_request_method,
+        canonical_uri,
+        canonical_querystring,
+        canonical_headers,
+        signed_headers,
+        hashed_request_payload
+    );
+    // println!("{}", canonical_request);
+
+    // step 2: 拼接待签名字符串
+    let credential_scope = format!("{}/{}/tc3_request", date, service);
+    let mut hasher = Sha256::new();
+    hasher.input_str(&canonical_request);
+    let hashed_canonical_request = hasher.result_str();
+    let string_to_sign = format!(
+        "{}\n{}\n{}\n{}",
+        algorithm, timestamp, credential_scope, hashed_canonical_request
+    );
+    // println!("{}", string_to_sign);
+
+    // step 3: 计算签名
+    fn sign<'a>(key: &'a [u8], msg: &'a [u8]) -> Vec<u8> {
+        let mut hmac = Hmac::new(Sha256::new(), key);
+        hmac.input(msg);
+        let result = hmac.result();
+        let code = result.code();
+        code.to_vec()
+    }
+    let key = "TC3".to_string() + secret_key;
+    let secret_date = sign(key.as_bytes(), date.as_bytes());
+    // println!("{}", secret_date.to_hex());
+    let secret_service = sign(&secret_date, service.as_bytes());
+    // println!("{}", secret_service.to_hex());
+    let secret_signing = sign(&secret_service, "tc3_request".as_bytes());
+    // println!("{}", secret_signing.to_hex());
+    let mut hmac = Hmac::new(Sha256::new(), &secret_signing);
+    hmac.input(string_to_sign.as_bytes());
+    let signature = hmac.result().code().to_hex();
```

Comment:
Maybe you can impl to_hex() yourself? Simply copy the function from rustc-serialize may be accepted.